fecgov/fec-cms
# Add vertical margin for buttons and org chart

## Summary
- Resolves #3259

Add 2.4 rem margin below internal/external buttons.
- Also added same margin to org chart. 
  - Since the org chart UL is floated left, it does not respond to spacing set via a utility class on immediate siblings. (ie: `u-margin...`, `u-padding...`).
  - Since its padding and margins are explicitly set to 0, it also does not respond to utility classes added  to itself. (ie: `<ul id="fec-orgchart" class="u-margin--bottom"...`

   ... so I explicitly set a bottom margin on it.

**Followup tasks:**
- [x] Remove extra spaces added in Wagtail to accommodate the previous lack of margin (Content team will hopefully be able to provide an inventory). **Followup Issue submitted by @dorothyyeager** : https://github.com/fecgov/fec-cms/issues/3387
- [x] Remind content team of utility classes available for use in  html blocks when existing styles don't provide the right vertical margins. See below:
 <details><summary>Utility Classes(Click to Expand...)</summary>
   
```
// Utility padding classes

.u-padding--top {
  padding-top: u(2rem);
}

.u-padding--top--small {
  padding-top: u(.5rem) !important;
}

.u-padding--bottom {
  padding-bottom: u(2rem);
}

.u-padding--bottom--small {
  padding-bottom: u(.5rem) !important;
}

.u-padding--left {
  padding-left: u(2rem);
}

.u-padding--right {
  padding-right: u(2rem);
}

.u-small-icon-padding--left {
  padding-left: u(2.5rem);
}

// Utility margin classes
.u-margin--bottom {
  margin-bottom: u(1rem);
}

.u-margin--bottom--small {
  margin-bottom: u(.5rem);
}

.u-margin--top {
  margin-top: u(1rem);
}

.u-margin--left {
  margin-left: u(1rem);
}

.u-margin--right {
  margin-right: u(1rem);
}

.u-negative--top--margin {
margin-top: u(-1rem);
}

```
</details>

## Impacted areas of the application
modified:   fec/static/scss/components/_fec-org-chart.scss
modified:   home/templates/home/full_width_page.html
modified:  fec/static/scss/components/_table-styles.scss
 

## How to test
**Content team:** Since this is probably best reviewed by content team, I will put this on feature so it can be tested in Wagtail.
**Developers:** Test on feature space or locally with these steps:
- checkout branch: `feature/padding-buttons-org-and-org-chart`
- `npm run build-sass`
- Check the org chart to make sure that it has spacing below it, before the first footnote (https://www.fec.gov/about/leadership-and-structure/fec-offices/)
- Edit this page and remove the extra paragraphs above headlines in rich-text areas that appear right after buttons. These were placed here to accommodate the lack of spacing. Before removing then you should now wee too much space.
   - This is the page: http://127.0.0.1:8000/help-candidates-and-committees/candidate-taking-receipts/partnership-llc-contributions/
   -  This is the quick link to edit it: http://127.0.0.1:8000/admin/pages/10547/edit/
- Publish or preview the above page and confirm that there is adequate spacing above/below buttons.
**Example of removing extra paragraph:**
![Nov-23-2019 01-23-47](https://user-images.githubusercontent.com/5572856/69474507-0a734680-0d90-11ea-9551-83db486c31e9.gif)

____
